the-collab-lab/tcl-12-smart-shopping-list
# Read/write to Firestore database

## Description



- Removed Create React App boilerplate
- Installed Firebase and Firestore dependencies
- Created a counter that increments by 1 with each button click
- Connected Firestore to store click count in database
- Update database count and have new count reflected on front end 

## Related Issue



Closes #1 

## Acceptance Criteria



* The following have been added as project dependencies: `firebase` & `react-firestore`
* Make a change in the Firestore database and it shows up in the app
* Make a change in the app and it shows up in the Firestore database

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
| ✓ | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before


![image](https://user-images.githubusercontent.com/61370161/95370916-1b8d4f00-089f-11eb-9110-021247e64bf3.png)


### After



![Screen Shot 2020-10-07 at 1 16 19 PM](https://user-images.githubusercontent.com/61370161/95371358-b84fec80-089f-11eb-9cbf-36f558e37dfc.png)

![Screen Shot 2020-10-07 at 1 16 35 PM](https://user-images.githubusercontent.com/61370161/95371376-bede6400-089f-11eb-96fd-7443bc4e6876.png)


## Testing Steps / QA Criteria


1.  From your terminal, pull down this branch (`git pull origin eb-al-connect-to-firestore`) and checkout to the branch (`git checkout eb-al-connect-to-firestore`)
2. Then install the newly added dependencies by running `npm install`
3.  Open up [Firestore Database](https://console.firebase.google.com/u/0/project/tcl-12-smart-shopping-list/firestore/data~2Fcount~2FFQyeXUHAfge9AXp6qXrK)
4. Click on the counter button and check to see if the count updates on the page and in the database.
5. Edit the count directly in the Firestore database by editing the number value and refresh the app. The count on the app should now reflect the value in the database.
